### ISSUE TITLE
doom upgrade: Repository "editorconfig-emacs" has a dirty worktree

### ISSUE THREAD

**Issue by rnadler** [editorconfig/editorconfig-emacs#358]
Every time I do a doom upgrade, it complains that `core-test` is dirty and requires manual intervention. It's been happening for the last month or two.

![image](https://github.com/user-attachments/assets/aa7e2229-1890-4384-baef-054912b3db82)

Not sure if this is a doom issue, but decided to start here.
Thanks!

**Comment by xuhdev** [comment]
I don't think it is related to this repo. Chances are you have made changes locally in the `core-test` submodule so it becomes dirty. Otherwise, doom may have done something. Could you double check your local dir or ask in their issue tracker?

**Comment by rnadler** [comment]
I think I found the issue. It looks like doom upgrade doesn't handle submodule updates properly. The 'dirty' content is due to this commit: https://github.com/editorconfig/editorconfig-emacs/commit/fbd078ad647888c971abe3508dd0ba27fac97cb9#diff-ca2843133fef83d92a33f4f1b36ad737ef7c239c0fcef39c182cd014f1dace83L1

This forced the manual removal of the old local submodule reference.

All I had to do was delete the straight repo to force a new clone. All is well ... until the next time the `core-test` reference is updated anyway.
